### DISCUSSION TITLE
Is 6.0 coming soon?, I just noticed latest version is  next: 7.0.0-next.1 before a 6.0.0 release

### DISCUSSION THREAD

**Discussion by bitfrost** [verdaccio/verdaccio#4018] [Q&A] [answered]
Master seems to moved to 7.0.0-next.* for versioning.  Is 6 getting ready for release?

**Comment by juanpicado** [comment]
Yeah probably October (If I don't find any blocker), I am preparing branch 6.x for it.

It will be same as v5 but with big dependency major changes and minor things. No breaking changes aside new Node.js minimums (probably 18) because v16 is out or maintenance in October this year.

What I was doing for 6.x has been rebranded to 7.x (why?) because supporting 5v was getting hard due nodejs 12 support so I decided move on with major versions like this and I want to minimize breaking changes near to 0.

**Reply by wongrichardalex** [reply]
Any updates on this? My team is looking to upgrade to 6.x due to a security vulnerability in the request dependency; when do you think 6 will be released now?

**Reply by juanpicado** [reply]
I was aiming this year but :) I'm almost on season holidays so let's see if I can make it (I never promise specifics dates because my time is variable) but you can follow progress here https://github.com/verdaccio/verdaccio/commits/6.x 

What's missing?
- Finish this PR https://github.com/verdaccio/verdaccio/pull/4113 (reuse code for easy upgrade and legacy signature support for easy migration)
- Search endpoint is not working correctly

(When I finish those topics, I will release beta releases for few weeks so you and others can start testing)

That's what's missing so far, aside of that 6.x E2E has been running here https://github.com/verdaccio/e2e-tests for a while without major issue.

**Reply by juanpicado** [reply]
Furthermore ... for eager testers what's new ... well ... I have a registry for canaries that you can pull (please don't abuse is just 5$ peanut server for build purposes)

```
npm install -g  verdaccio@canary-6x --registry https://registry.verdaccio.org
```

or docker

```
docker pull verdaccio/verdaccio:6.x-next
```

That would be enough for quick testing.

**Reply by wesleytodd** [reply]
I would love to test it out, but will not be installing from the verdaccio registry since I am in an internal project configured to load from our registry proxy. Is there a version published to the public registry?

Should I just install `6.0.0-6-next.76` directly since that seems published to the public registry?

**Reply by juanpicado** [reply, marked as answer]
ℹ️ EDIT: check https://github.com/verdaccio/verdaccio/discussions/4018#discussioncomment-8036037


> Is there a version published to the public registry?

No yet, I know is confusing the `6.0.0-6-next.76` is not 6 anymore (why? :) because it took me too much time to publish and I had to  rename to v7 due Node.js 12 was though to keep supported and that version you shared is same as https://www.npmjs.com/package/verdaccio/v/7.0.0-next.4 which is under the dist tag `next` `npm install -g verdaccio@next` (a.k.a) next future version that allows me to experiment safely.

I will publish 6.x branch under some dist-tag candidate release when I feel confident do not break anything, hopefully the next month :) when I back from open source holidays, by Q1 2024 probably release 6.x definitely and start deprecating 5.x 6 months from the 6.x release, no big breacking changes expected.

**Reply by wesleytodd** [reply]
Ahh, ok interesting. I was wondering when I noticed that the repo was on 7 and I had installed 5 as `latest` in my project just now. Thanks for the context.  I can test out the `next` tag then, just wanted to make sure I was looking at the right thing since your instructions above didn't not include a public registry install instruction.

**Reply by juanpicado** [reply]
To avoid confusion, https://www.npmjs.com/package/verdaccio/v/6.0.0-6-next.76?activeTab=dependents is deprecated

**Reply by juanpicado** [reply]
fyi @wongrichardalex `request` was just removed from latest v5 https://github.com/verdaccio/verdaccio/releases/tag/v5.29.0 kudos to @legobeat

**Reply by wongrichardalex** [reply]
WHOOOOOO!!!!! Thank you.

**Comment by dnafication** [comment]
@juanpicado Are there any server-related issues which needs help for this release? I have allocated some time during Hacktoberfest to make contributions, whether it be coding, reviewing, or testing. Additionally, how can I filter which issues are relevant to each version i.e. v5 or v5?

**Reply by juanpicado** [reply]
@dnafication you can check https://github.com/verdaccio/verdaccio/discussions/1461 or if you have follow up questions feel free to drop them there

**Comment by juanpicado** [comment]
You can start testing via npmjs https://www.npmjs.com/package/verdaccio/v/6.0.0-beta.1

```bash
npm install -g verdaccio@latest-6
```

**Comment by juanpicado** [comment]
ℹ️  v6.0.0-beta.2 released

https://github.com/verdaccio/verdaccio/releases/tag/v6.0.0-beta.2

![Kapture 2024-03-17 at 11 43 06](https://github.com/verdaccio/verdaccio/assets/558752/a21537a1-794d-484a-8d10-562af119f679)

**Comment by viceice** [comment]
There's now a v6 release, are there any upgrade nodes to upgrade from v5?

**Reply by juanpicado** [reply]
Please check out the release notes, details are there.

**Reply by danopia** [reply]
Release notes: https://github.com/verdaccio/verdaccio/releases/tag/v6.0.0
